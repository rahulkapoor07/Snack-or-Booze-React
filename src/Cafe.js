import React, { useState, useEffect } from "react";
// import { BrowserRouter } from "react-router-dom";

import SnackOrBoozeApi from "./Api";
import cafeContext from "./cafeContext";

//I have created a seperate component with name Cafe to store all the snacks state and
// drinks states.


function Cafe({children}) {
  const [isLoading, setIsLoading] = useState(true);
  const [snacks, setSnacks] = useState([]);
  const [drinks, setDrinks] = useState([]);

  useEffect(() => {
    async function getSnacks() {
      let snacks = await SnackOrBoozeApi.getSnacks();
      setSnacks(snacks);
      setIsLoading(false);
    }
    getSnacks();
  }, []);

  useEffect(() => {
    async function getDrinks() {
      let drinks = await SnackOrBoozeApi.getDrinks();
      setDrinks(drinks);
      setIsLoading(false);
    }
    getDrinks();
  }, []);

  //This function will collect data from Form and add that data into snacks state or drinks state depends
  // upon the type/
  const collectFormData = ({type, name, description, recipe, serve})=>{
    type === "Snacks" ? setSnacks(data => ([...data, {id : name, name, description, recipe, serve}])) :
      setDrinks(data => ([...data, {id : name, name, description, recipe, serve}]))
  }

  if (isLoading) {
    return <p>Loading &hellip;</p>;
  }

  // I have create a context with name cafeContext get pass some values of drinks, snacks and
  // collectData function to Route component where I have put all the routes with thier 
  // respective components.
  return(
    <cafeContext.Provider value={{drinks, snacks, collectFormData}}>

    {children}

  </cafeContext.Provider>
  )
}

export default Cafe;