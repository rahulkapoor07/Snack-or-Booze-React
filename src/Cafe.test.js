import React from "react";
import {MemoryRouter} from "react-router-dom";
import { render } from '@testing-library/react';
import Cafe from './Cafe';


describe("Testing Cafe component", ()=>{
test("render", ()=>{
    render(<MemoryRouter><Cafe /></MemoryRouter>)
});

test("SnapShot test", ()=>{
    const {asFragment}= render(<MemoryRouter><Cafe /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
});
});