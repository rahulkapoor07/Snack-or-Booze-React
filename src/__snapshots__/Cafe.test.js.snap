// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot test 1`] = `
<DocumentFragment>
  <p>
    Loading …
  </p>
</DocumentFragment>
`;

exports[`Testing Cafe component SnapShot test 1`] = `
<DocumentFragment>
  <p>
    Loading …
  </p>
</DocumentFragment>
`;
